Allow using default gRPC dialer

brandonSc opened the pull request:
The builtin dialer in gRPC has support for things like HTTP connect proxy which we need to support BYOC satellites in some environments. Buildkit was previously forcing it's own dialer implementation.

mikejholly reviewed (approved):
Makes sense. Code looks good!

brandonSc commented:
Corresponding earthly PR https://github.com/earthly/earthly/pull/4115

brandonSc committed:
feat: allow option for client to use grpc dialer

Buildkit has a custom `resolveDialer` dailer, that does not support HTTP proxy.
This adds the ability for a client to enable the default gRPC dialer, which
allows support for HTTP.

alexcb reviewed (approved):
looks good; thanks for tidying up the commits.
